File: app/src/main/java/mattframpton/endclothing_mattframpton/adapter/ProductCatalogueAdapter.java
package mattframpton.endclothing_mattframpton.adapter;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.bumptech.glide.Glide;

import java.util.ArrayList;

import mattframpton.endclothing_mattframpton.model.Product;
import mattframpton.endclothing_mattframpton.R;

public class ProductCatalogueAdapter extends RecyclerView.Adapter<ProductCatalogueHolder>{

	ArrayList<Product> productList;

	public ProductCatalogueAdapter(ArrayList data){
		this.productList = data;
	}

	@NonNull
	@Override
	public ProductCatalogueHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
		View v = LayoutInflater.from(parent.getContext()).inflate(R.layout.lineitem, parent, false);
		return new ProductCatalogueHolder(v);
	}

	@Override
	public void onBindViewHolder(@NonNull ProductCatalogueHolder holder, int position) {
		holder.productPrice.setText(productList.get(position).getPrice());
		holder.productName.setText(productList.get(position).getName());
		Glide.with(holder.productImage.getContext())
				.load(productList.get(position).getImage())
				.into(holder.productImage);
	}

	@Override
	public int getItemCount() {
		return productList.size();
	}
}

File: app/src/main/java/mattframpton/endclothing_mattframpton/adapter/ProductCatalogueAdapter.kt
package mattframpton.endclothing_mattframpton.adapter

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.bumptech.glide.Glide
import mattframpton.endclothing_mattframpton.R
import mattframpton.endclothing_mattframpton.model.Product
import java.util.*

class ProductCatalogueAdapter(data: ArrayList<*>) : RecyclerView.Adapter<ProductCatalogueHolder>() {

    internal var productList: ArrayList<Product>

    init {
        this.productList = data as ArrayList<Product>
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ProductCatalogueHolder {
        val v = LayoutInflater.from(parent.context).inflate(R.layout.lineitem, parent, false)
        return ProductCatalogueHolder(v)
    }

    override fun onBindViewHolder(holder: ProductCatalogueHolder, position: Int) {
        holder.productPrice.text = productList[position].price
        holder.productName.text = productList[position].name
        Glide.with(holder.productImage.context)
                .load(productList[position].image)
                .into(holder.productImage)
    }

    override fun getItemCount(): Int {
        return productList.size
    }
}
